File: 06_ListView_Example/app/src/main/java/com/example/listview_example/MainActivity.java
package com.example.listview_example;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.BaseAdapter;
import android.widget.ListView;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    String arr[] = new String[]{
            "Iron Man",
            "Spider Man",
            "Doctor Strange",
            "Wanda Vision",
            "Captian Marvel",
            "Caption America",
            "Black Pather",
            "Star Lord",
            "Gamora",
            "Thor",
            "Hulk",
            "Loki",
            "Black Widow",
            "Ant Man",
            "Wasp",
            "Falcon",
            "Winter Soilder"
    };

    POJOModal list[] = POJOModal.generate10Model().toArray(new POJOModal[0]);

    ListView listVw;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        listVw = findViewById(R.id.listView);

        // Using Default Adaptor
//        ArrayAdapter adptr = new ArrayAdapter<String>(
//                this,
//                android.R.layout.simple_list_item_1,
//                android.R.id.text1,
//                arr
//        );
//        listVw.setOnItemClickListener(new AdapterView.OnItemClickListener() {
//                                          @Override
//                                          public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
//                                              Toast.makeText(MainActivity.this, String.format("%s %s", position, arr[position]), Toast.LENGTH_SHORT).show();
//                                          }
//                                      }
//        );
//        listVw.setAdapter(adptr);


        //Creating Custom Adaptor
        listVw.setAdapter(new CourseAdaptor());
    }

    class CourseAdaptor extends BaseAdapter {

        @Override
        public int getCount() {
            return list.length;
        }

        @Override
        public POJOModal getItem(int position) {
            return list[position];
        }

        @Override
        public long getItemId(int position) {
            return 0;
        }

        @Override
        public View getView(int position, View convertView, ViewGroup parent) {

            View viewItem = getLayoutInflater().inflate(
                    R.layout.list_of_courses,
                    parent,
                    false
            );

            TextView name = viewItem.findViewById(R.id.name);
            TextView course = viewItem.findViewById(R.id.course);

            name.setText(getItem(position).getName());
            course.setText(getItem(position).getCourse());

            return  viewItem;
        }
    }
}

File: 11_TodoApp_SQLite/app/src/main/java/com/example/todoapp/MainActivity.kt
package com.example.todoapp

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import androidx.databinding.DataBindingUtil
import com.example.todoapp.databinding.ActivityMainBinding
import com.example.todoapp.db.MyDbHelpe
import com.example.todoapp.db.TodoTable

class MainActivity : AppCompatActivity() {

    var todos = ArrayList<Todo>()
    lateinit var mBinding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
      mBinding=DataBindingUtil.setContentView(this,R.layout.activity_main)

        val listAdaptor = ArrayAdapter<Todo>(
            this,
            android.R.layout.simple_list_item_1,
            android.R.id.text1,
            todos
        )

     //   val listView = findViewById<ListView>(R.id.lstView)
       // val btn = findViewById<Button>(R.id.btn)
        //val editText = findViewById<EditText>(R.id.editText)
        mBinding.btn.setOnClickListener {

        }
//        listView.adapter = listAdaptor
        val todoAdaptor = TodoAdaptor( this,todos)
        mBinding.lstView.adapter = todoAdaptor

        val db = MyDbHelpe(this).writableDatabase

        fun refreshTodoList() {
            todos.clear()
            todos.addAll(TodoTable.getAllTodos(db))
            Log.d("TODOS", todos.toString())
            todoAdaptor.notifyDataSetChanged()
//            listAdaptor.notifyDataSetChanged()
        }
        refreshTodoList()

        mBinding.btn.setOnClickListener {
            val newTodo = Todo(
                mBinding.editText.text.toString(),
                false
            )
            TodoTable.insertTodo(db, newTodo)
            mBinding.editText.setText("")

            refreshTodoList()

        }
    }
    class TodoAdaptor(val context: Context,val todos: ArrayList<Todo>,
                       ) : BaseAdapter() {
        override fun getCount(): Int {
            return  todos.size
        }

        override fun getItem(position: Int): Any {
            return todos.get(position)
        }

        override fun getItemId(position: Int): Long {
            return 0
        }

        override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {

            val view = LayoutInflater.from(context).inflate(
                android.R.layout.simple_list_item_1,
                parent,false
            )
            view.findViewById<TextView>(android.R.id.text1).setText(todos.get(position).task)

            return view
        }

    }
}